include/nifty/graph/label_propagation.hxx

```diff
@@ -0,0 +1,124 @@
+//
+// Created by Alberto Bailoni on 07/01/2021.
+//
+
+#pragma once
+
+#include <functional>
+#include <set>
+#include <vector>
+#include <unordered_set>
+#include <boost/container/flat_set.hpp>
+#include <string>
+#include <cmath>        // std::abs
+
+// #include "nifty/tools/changable_priority_queue.hxx"
+// #include "nifty/graph/edge_contraction_graph.hxx"
+// #include "nifty/graph/agglo/cluster_policies/cluster_policies_common.hxx"
+#include <iostream>
+#include "xtensor/xarray.hpp"
+
+#include "nifty/parallel/threadpool.hxx"
+#include <ctime> // Random seed for shuffle
+#include <xtensor/xrandom.hpp>
+
+
+namespace nifty{
+    namespace graph {
+        template<class GRAPH, class LABELS, class SIGNED_WEIGHTS, class IS_LOCAL_EDGE>
+        void runLabelPropagation(const GRAPH & graph,
```

Comment:
Could you add some description on what this does exactly and add a test for it (in python is fine of course.)

Reply:
I added it now